pkg/controller/servicebinding/binder.go

```diff
@@ -345,14 +352,18 @@ func (b *binder) appendEnvVar(
 // appendEnvFrom based on secret name and list of EnvFromSource instances, making sure secret is
 // part of the list or appended.
 func (b *binder) appendEnvFrom(envList []corev1.EnvFromSource, secret string) []corev1.EnvFromSource {
-	for _, env := range envList {
-		if env.SecretRef != nil && env.SecretRef.Name == secret {
-			b.logger.Debug("Directive 'envFrom' is already present!")
-			// secret name is already referenced
-			return envList
+	for k, env := range envList {
+		if env.SecretRef != nil {
+			if env.SecretRef.Name == secret {
+				b.logger.Debug("Directive 'envFrom' is already present!")
+				// secret name is already referenced
+				return envList
+			} else if strings.Contains(env.SecretRef.Name, b.sbr.GetName()) {
```

Comment:
Contains is error prone. Check for equality.

Reply:
Done